pkg/search/controllers_test.go

```diff
@@ -577,11 +584,12 @@ func TestDeleteResourceRegistryEventHandler(t *testing.T) {
 // createController initializes a new Controller instance using the provided
 // Kubernetes REST configuration, shared informer factory, and REST mapper.
 // It returns the created Controller or an error if initialization fails.
-func createController(restConfig *rest.Config, factory informerfactory.SharedInformerFactory, restMapper meta.RESTMapper) (*Controller, error) {
+func createController(restConfig *rest.Config, factory informerfactory.SharedInformerFactory, restMapper meta.RESTMapper, stopCh chan struct{}) (*Controller, error) {
 	newController, err := NewController(restConfig, factory, restMapper)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create new controller, got: %v", err)
 	}
+	newController.InformerManager = genericmanager.NewMultiClusterInformerManager(stopCh)
```

Comment:
Is this modification unnecessary? newController.InformerManager has been initialized.

Reply:
The new multicluster informer manager global instance is created using the singleton pattern (`once`), which ensures it is instantiated only once. As a result, test cases depend on this single instance, leading to unintended dependencies between the tests. What do you think?

https://github.com/karmada-io/karmada/blob/8d89d77110cf378d09b5f43dcb2e09bb393d61f1/pkg/util/fedinformer/genericmanager/multi-cluster-manager.go#L39-L44

Reply:
Yes, you're right, it's right not to share data between each test case.